src/algebra/big_operators.lean

```diff
@@ -546,11 +555,23 @@ attribute [to_additive sum_smul'] prod_pow
 @prod_const _ (multiplicative β) _ _ _
 attribute [to_additive] prod_const
 
+lemma sum_const_on_nat {m : ℕ} {f : α → ℕ} (h₁ : ∀x ∈ s, f x = m) :
+  s.sum f = card s * m :=
+begin
+  rw [← nat.smul_eq_mul, ← sum_const],
+  apply sum_congr rfl h₁
+end
+
 lemma sum_range_succ' [add_comm_monoid β] (f : ℕ → β) :
   ∀ n : ℕ, (range (nat.succ n)).sum f = (range n).sum (f ∘ nat.succ) + f 0 :=
 @prod_range_succ' (multiplicative β) _ _
 attribute [to_additive] prod_range_succ'
 
+lemma sum_flip [add_comm_monoid β] {n : ℕ} (f : ℕ → β) :
+  sum (range (n+1)) (λ r, f (n - r)) = sum (range (n+1)) f :=
```

Comment:
You have `n+1` in this lemma, and `n.succ` in the multiplicative version. I don't know which one is favored in this file, but it would be good to enforce as much coherence as possible.